source/shared/core_stmt.cpp

```diff
@@ -2662,25 +2661,29 @@ bool reset_ae_stream_cursor( _Inout_ sqlsrv_stmt* stmt ) {
 }
 
 void adjustInputPrecision( _Inout_ zval* param_z, _In_ SQLSMALLINT decimal_digits ) {
-    // 38 is the maximum precision supported for sql decimal types
-    // 6 is a composition of: 1 for '.'; 1 for sign of the number;
-    //   1 for 'e' or 'E' (scientific notation); 1 for sign of scientific exponent; 2 for length of scientific exponent
-    size_t maxDecimalLen = 38 + 6;
-    SQLSRV_ASSERT( Z_STRLEN_P( param_z ) < maxDecimalLen, "Input decimal overflow: sql decimal type only supports up to a precision of 38." );
+    size_t maxDecimalPrecision = 38;
+    // maxDecimalStrLen is the maximum length of a stringified decimal number
+    // 6 is derived from: 1 for '.'; 1 for sign of the number; 1 for 'e' or 'E' (scientific notation);
+    //                    1 for sign of scientific exponent; 2 for length of scientific exponent
+    // if the length is greater than maxDecimalStrLen, do not change the string
+    size_t maxDecimalStrLen = maxDecimalPrecision + 6;
+    if (Z_STRLEN_P(param_z) > maxDecimalStrLen) {
+        return;
+    }
     std::vector<size_t> digits;
     char* ptr = ZSTR_VAL( Z_STR_P( param_z ));
     bool isNeg = false;
     char scientificChar = ' ';
-    int scientificExp = 0;
+    short scientificExp = 0;
     // parse digits in param_z into the vector digits
     if( *ptr == '+' || *ptr == '-' ){
-        if( *ptr = '-' ){
+        if( *ptr == '-' ){
             isNeg = true;
```

Comment:
this seems to suggest you have assumed no space characters before the positive / negative sign?